lib/cover_my_meds/client/forms.rb

```diff
@@ -5,14 +5,28 @@ module Forms
     CURRENT_VERSION = 1
 
     def form_search form, drug_id, state, version=CURRENT_VERSION
-      params = {q: form, drug_id: drug_id, state: state, v: version}
-      data = forms_request GET, params: params
-      data['forms'].map { |d| Hashie::Mash.new(d) }
+      search_with_args({q: form}, drug_id, state, version)
+    end
+
+    def form_search_by_plan plan_hash, drug_id, state, version=CURRENT_VERSION
+      search_with_args(plan_hash, drug_id, state, version)
     end
 
     def get_form form_id, version = CURRENT_VERSION
       data = forms_request GET, params: { v: version }, path: form_id
       Hashie::Mash.new(data['form'])
     end
+
+    private
+
+    def search_with_args(query_args, drug_id, state, version)
```

Comment:
I did, but it would actually be duplicate code because the logic for turning the drug_id/state/version into a params hash would appear in two places.  I wasn't thrilled with the duplicate args in the method signatures, but it seemed like the lesser of several evils.